Portal/vip-gatelab/src/main/java/fr/insalyon/creatis/vip/gatelab/client/GateLabModule.java

```diff
@@ -47,18 +47,24 @@
 public class GateLabModule extends Module {
 
     public GateLabModule() {
-      new ApplicationModule();
+      //new ApplicationModule();
     }
 
     @Override
     public void load() {
-
+        
         if (CoreModule.user.isSystemAdministrator()
                 || CoreModule.user.hasGroupAccess(GateLabConstants.GROUP_GATELAB)) {
-            for(String GateLabClass:ApplicationModule.reservedClasses){
+            for(String GateLabClass:ApplicationModule.reservedGateClasses){
             CoreModule.addApplicationsTileGrid(new GateLabTileGrid(GateLabClass));
             }
         }
+        if (CoreModule.user.isSystemAdministrator()
+                || CoreModule.user.hasGroupAccess(GateLabConstants.GROUP_GATELABTEST)) {
+            for(String GateLabClass:ApplicationModule.reservedTestClasses){
+                CoreModule.addApplicationsTileGrid(new GateLabTileGrid(GateLabClass));
+            }
+        }
```

Comment:
As a general comment, I think a cleaner way to implement this would be:
1. to add a "gateClasses" boolean to ApplicationService.Util.getInstance().getClasses to retrieve either the general classes or the gate classes. Only the classes that the user can access are returned (as in the existing implementation).
2. to implement ApplicationModule.load and GateLabModule.load in the same way, calling ApplicationService.Util.getInstance().getClasses with "true" or "false" for "gateClasses".

That would get rid of the reservedClass buggy code in the ApplicationModule constructor (see comment above). It would also do security checks server side rather than client side.